### ISSUE TITLE
Sequel (and other db support) for transactions

### ISSUE THREAD

**Issue by Aryk** [aasm/aasm#474]
I noticed AR was the only supported database for transactions.

You guys cool if I abstract this method out into a 'DatabasePersistence' module and then you would just need to define the transaction wrapper method? Then it could be easily used with Sequel.

```ruby
        def aasm_fire_event(state_machine_name, name, options, *args, &block)
          event = self.class.aasm(state_machine_name).state_machine.events[name]

          if options[:persist]
            event.fire_callbacks(:before_transaction, self, *args)
            event.fire_global_callbacks(:before_all_transactions, self, *args)
          end

          begin
            success = if options[:persist]
              self.class.transaction(:requires_new => requires_new?(state_machine_name)) do
                lock!(requires_lock?(state_machine_name)) if requires_lock?(state_machine_name)
                super
              end
            else
              super
            end

            if options[:persist] && success
              event.fire_callbacks(:after_commit, self, *args)
              event.fire_global_callbacks(:after_all_commits, self, *args)
            end
          ensure
            if options[:persist]
              event.fire_callbacks(:after_transaction, self, *args)
              event.fire_global_callbacks(:after_all_transactions, self, *args)
            end
          end

          success
        end
```

Then it could be potentially used for more orms in aasm.

LMK, don't want to do a PR if it's unlikely to be merged...

**Comment by anilmaurya** [comment]
I liked your Idea, go ahead with pull request 👍

**Comment by Aryk** [comment]
PR sent!

**Comment by anilmaurya** [comment]
Fixed with #475
